main/main.cpp

```
@@ -9,48 +9,68 @@
+
     /* Initialize the i2c bus for the current platform */
-    sensirion_i2c_init();
+    if (sensor.initializeBus() == false) {
+        printf("Failed to initialize I2C bus. Exiting...\n");
+        return;
+    }
+
+/* Probe loop with backoff delay to yield CPU, limited retries */
+int retry_count = 0;
+const int max_retries = 10;
+int delay_ms = 50;
```

Clang (14.0.6)

[warning] 50 is a magic number; consider replacing it with a named constant

(cppcoreguidelines-avoid-magic-numbers,readability-magic-numbers)